lib/infomon.rb

```diff
@@ -0,0 +1,515 @@
+
+    @save_value = {}
+    $infomon_values = DB_Store.read('infomon')
+    pp $infomon_values if $infomon_debug
+    # sleep 0.5
+    respond 'starting sequence' if $infomon_debug
+    respond $infomon_values if $infomon_debug
+    echo $infomon_values['active_spells'] if $infomon_debug
+    echo $infomon_values['need_psm_update'] if $infomon_debug
+    echo $infomon_values['Stats'] if $infomon_debug
+
+    respond 'Calling active spells' if $infomon_debug
+    $infomon_values['active_spells'] = {} unless $infomon_values['active_spells'].instance_of?(Hash)
+
+    respond 'Calling need psm update' if $infomon_debug
+    $infomon_values['need_psm_update'] = true if $infomon_values['need_psm_update'].nil?
+
+    #
+    # Load spell info
+    #
+    unless Spell.load
+      echo 'error: failed to load spell list'
+      exit
+    end
+
+    #
+    # Load or get character information
+    #
+
+    psm_abilities = if Gem::Version.new(LICH_VERSION) < Gem::Version.new('5.0.16')
+                      ['CMan']
+                    else
+                      %w[Feat Armor Weapon Shield CMan]
+                    end
+    need_psm = []
+    if $infomon_values['need_psm_update']
+      need_psm = psm_abilities.dup
+      $infomon_values['need_psm_update'] = false
+    else
+      psm_abilities.each do |ability|
+        if !defined?(Kernel.const_get(ability.to_s))
+          pp 'Encountered fatal error'
+          exit
+        elsif $infomon_values[ability.to_s.downcase].nil?
+          $infomon_values[ability.to_s.downcase] = {}
+          need_psm << ability
+        else
+          begin
+            unless $infomon_values[ability.to_s.downcase].empty?
+              $infomon_values[ability.to_s.downcase].each_pair { |psm, rank| Kernel.const_get(ability.to_s).send("#{psm}=", rank) }
+            end
+          rescue StandardError
+            pp 'Bad juju happened here.'
+            nil
+          end
+        end
+      end
+    end
+
+    silence_me
+    echo 'checking psms...' unless need_psm.empty?
+    need_psm.each do |get_ability|
+      Lich::Statsinfo.request('psms', get_ability)
+    end
+    silence_me
+
+    respond 'calling Stats, Skills, Spells, Society, Citizenship' if $infomon_debug
+    if $infomon_values['Stats'] &&
+       $infomon_values['Skills'] &&
+       $infomon_values['Spells'] &&
+       $infomon_values['Society'] &&
+       $infomon_values['citizenship']
+
+      begin
+        respond 'Now LOADING Stats, Skills, Spells, Society, Citizenship' if $infomon_debug
+        Stats.load_serialized   = $infomon_values['Stats']
+        Skills.load_serialized  = $infomon_values['Skills']
+        Spells.load_serialized  = $infomon_values['Spells']
+        Society.load_serialized = $infomon_values['Society']
+        Char.citizenship        = $infomon_values['citizenship']
+      rescue StandardError
+        echo $ERROR_INFO
+        echo $ERROR_INFO.backtrace[0..1]
+        exit
```

Comment:
This will log you out of the game with no info

Reply:
Scoped the library back to being required to be loaded from a script, `infomon.lic` for now. Which allows for safe `exit` on this failure.